### PR TITLE
feat: yearn market service

### PR DESCRIPTION
**Pull request by theoboldfrazier**
## Description

Updates files for new types and utilizes new market service signatures.

## Notice

Before submitting a pull request, please make sure you have answered the following:

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?
- [x] Do all new and existing tests pass? Does the linter pass?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

If applicable, please link to the github issue and put `closes #XXXX` in your comment to auto-close the issue that your PR fixes.

## Testing

Please outline all testing steps

1. Pull branch locally and run `yarn` to install new deps
2. etc...

### CODE REVIEW

**Review by 0xdef1cafe — APPROVED** [review]
preemptive approval

File: headers.js
Comment (reidrankin) [inline headers.js:16]:
```suggestion
    'https://test-api.yearn.network/v1/',
    'https://api.thegraph.com/subgraphs/name/salazarguille/yearn-vaults-v2-subgraph-mainnet/',
    'https://raw.githubusercontent.com/yearn/yearn-assets/',
```
I think this slightly-simplified version would make sense, though I do wonder if `test-api.yearn.network` is the right endpoint. Will this change in prod, and where does the URL come from? If it's configurable, it might be better to pull it in from `.env` or whatever.

Note that if a path is included in a CSP directive, a trailing slash means it's a prefix, whereas no trailing slash requires an exact match. I've added a trailing slash to the `api.thegraph.com` as sort of a guess here, since loading that page redirects to `https://api.thegraph.com/subgraphs/name/salazarguille/yearn-vaults-v2-subgraph-mainnet/graphql` in my browser, but double-check me on that.

**Review by reidrankin — CHANGES_REQUESTED** [review]
I have no complaints about the changeset itself, but there are a couple of security things we need to check before we merge this, so I'm marking this as "changes requested" to make sure that it's clear that we need to not merge before these get verified:
- We need to double-check that `yearn.network` is run by the same people as `yearn.finance` before we allow it in the CSP... is this API endpoint coming from an official client or something like that?
- I note that the [repo](https://github.com/salazarguille/yearn-vaults-v2-subgraph) for that subgraph seems to indicate that it's non-official; what's this used for, and are we sure the data's trustworthy?

File: headers.js
Comment (theoboldfrazier) [reply headers.js]: Great questions @reidrankin. For the question of yearn.network. All of these urls came from network calls when using the yearn-sdk. The yearn.network url is hardcoded in their sdk here https://github.com/yearn/yearn-sdk/blob/master/src/context.ts on line 66. From what I can tell, that is from the official yearn github account that is linked to from yearn.finance.

For the question about the subgraph url, that one also came from their sdk here: https://github.com/yearn/yearn-sdk/blob/master/src/services/subgraph/index.ts on line 16. The official yearn github account forked the salazarguille repo here https://github.com/yearn/yearn-vaults-v2-subgraph.

For the githubusercontet url, that was just for asset icons.

As for the data itself. The data from these endpoints used by our app is for calculating and displaying pricing data for yearn assets. So though our app doesn't source _balances_ for yearn assets from these, it does use the balance from unchained in combination with this data to calculate the price in USD displayed and the price history charts. So, assuming maliciousness from these data sources, the effect to the user could be that they see a different price value for their balance on yearn assets, though their underlying asset balance would still be from unchained.

Comment (theoboldfrazier) [reply headers.js]: For whatever reason, the trailing slash on the subgraph url does break the call to get historical pricing data. So I removed it.

**Review by elmutt — APPROVED** [review]

File: headers.js
Comment (reidrankin) [reply headers.js]: Verified -- and sorry for the delay!

(Looking at the Yearn SDK more generally, they _really_ like hardcoding URLs to random crap, so I doubt there's a scalable solution here...)

**Review by reidrankin — APPROVED** [review]

**Pull request merged**
